### PR TITLE
rgw: Add support bucket policy for subuser

### PR DESCRIPTION
**Pull request by clwluvw**
Add support bucket policy for subuser

Fixes: https://tracker.ceph.com/issues/39605
Signed-off-by: Seena Fallah <seenafallah@gmail.com>

### CODE REVIEW

**Comment by clwluvw** [comment]
@mattbenjamin I have updated bucketpolicy doc.

**Comment by clwluvw** [comment]
jenkins test make check

File: src/rgw/rgw_auth.cc [hunk 1]
```diff
@@ -617,6 +617,14 @@ bool rgw::auth::LocalApplier::is_identity(const idset_t& ids) const {
 	       (id.get_id() == user_info.user_id.id)) {
       return true;
     }
+    for (auto subuser : user_info.subusers) {
+      std::string user = user_info.user_id.id;
+      user.append(":");
+      user.append(subuser.second.name);
+      if (user == id.get_id()) {
+        return true;
+      }
```
Comment (pritha-srivastava) [inline src/rgw/rgw_auth.cc]: I think there should be a check for tenant also here, similar to the tenant check for a user.

Comment (clwluvw) [reply src/rgw/rgw_auth.cc]: Done :heavy_check_mark:

**Comment by ivancich** [comment]
jenkins test make check

**Comment by LenzGr** [comment]
Just wondering: will this new functionality be available via the Admin Ops API as well? Does the Dashboard need to be adapted to support this feature?

**Comment by clwluvw** [comment]
@adamemerson @ivancich Tests has been added.

**Comment by clwluvw** [comment]
@ivancich Can we have it in next nautilus release? :)

**Comment by ivancich** [comment]
@clwluvw Due to your comment I've updated the tracker to have backports to nautilus and octopus.

@cbodley @adamemerson Any objection?

**Comment by adamemerson** [comment]
No objections. It's a very small patch.

**Comment by ivancich** [comment]
Thanks, @adamemerson.

**Comment by clwluvw** [comment]
@ivancich Should I cherry pick it by my self to nautilus branch?

**Comment by ivancich** [comment]
@clwluvw We do have a backport team that will create the nautilus backport PR and associated tracker, get the reviews, run it through QA, and ultimately merge. When I've tried to do it myself, I often don't do it the way the backport team likes it done, and @smithfarm asks me for adjustments.

If you're familiar with their process or are willing to work with @smithfarm to get it right, then you can create the PR and tracker.

**Comment by clwluvw** [comment]
@ivancich I have seen the [backport tracker](https://tracker.ceph.com/issues/44146) but I don't see the PR for that. I thought I should do it by my self. I will wait for @smithfarm for cherry pick this PR.
Thanks. :)

**Pull request merged**